src/modules/mkNakedShell.nix

```diff
@@ -0,0 +1,136 @@
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+{ bashInteractive
+, coreutils
+, system
+, writeTextFile
+, lib
+}:
+let
+  bashPath = "${bashInteractive}/bin/bash";
+  stdenv = writeTextFile {
+    name = "naked-stdenv";
+    destination = "/setup";
+    text = ''
+      # Fix for `nix develop`
+      : ''${outputs:=out}
+
+      runHook() {
+        eval "$shellHook"
+        unset runHook
+      }
+    '';
+  };
+in
+{ name
+, # A path to a buildEnv that will be loaded by the shell.
+  # We assume that the buildEnv contains an ./env.bash script.
+  profile
+, env ? { }
+, shellHook ? ""
+, meta ? { }
+, passthru ? { }
+}:
+let
+  # simpler version of https://github.com/numtide/devshell/blob/20d50fc6adf77fd8a652fc824c6e282d7737b85d/modules/env.nix#L41
+  envToBash = name: value: "export ${name}=${lib.escapeShellArg (toString value)}";
+  startupEnv = lib.concatStringsSep "\n" (lib.mapAttrsToList envToBash env);
+  derivationArg = {
+    inherit name system;
+
+    # `nix develop` actually checks and uses builder. And it must be bash.
+    builder = bashPath;
+
+    # Bring in the dependencies on `nix-build`
+    args = [ "-ec" "${coreutils}/bin/ln -s ${profile} $out; exit 0" ];
+
+    # $stdenv/setup is loaded by nix-shell during startup.
+    # https://github.com/nixos/nix/blob/377345e26f1ac4bbc87bb21debcc52a1d03230aa/src/nix-build/nix-build.cc#L429-L432
+    stdenv = stdenv;
+
+    # The shellHook is loaded directly by `nix develop`. But nix-shell
+    # requires that other trampoline.
+    shellHook = ''
+      # Remove all the unnecessary noise that is set by the build env
+      unset NIX_BUILD_TOP NIX_BUILD_CORES NIX_STORE
+      unset TEMP TEMPDIR TMP TMPDIR
+      # $name variable is preserved to keep it compatible with pure shell https://github.com/sindresorhus/pure/blob/47c0c881f0e7cfdb5eaccd335f52ad17b897c060/pure.zsh#L235
+      unset builder out shellHook stdenv system
+      # Flakes stuff
+      unset dontAddDisableDepTrack outputs
+
+      # For `nix develop`. We get /noshell on Linux and /sbin/nologin on macOS.
+      if [[ "$SHELL" == "/noshell" || "$SHELL" == "/sbin/nologin" ]]; then
+        export SHELL=${bashPath}
+      fi
+
+      # https://github.com/numtide/devshell/issues/158
+      PATH=''${PATH#/path-not-set:}
+
+      export DEVENV_PROFILE=${profile}
```

Comment:
This should probably be documented, question is where.
This allows easy access to the built environment for end users to refer to in their scripts.

Reply:
https://devenv.sh/files-and-variables/